### ISSUE TITLE
ACE compilation with BuildPackages.sh gives a lot of error message

### ISSUE THREAD

**Issue by hungaborhorvath** [gap-packages/ace#6]
I obtain the following error messages during compiling by ````BuildPackages```` (GAP 4.8.6 stable downloaded from the website)), and I am wondering if these really should occur. The package seems to load ok in GAP, though.
````
==== Checking ace-5.2
../../sysinfo.gap found
Makefile successfully created.
Done. Now call 'make'
mkdir -p bin/x86_64-pc-linux-gnu-gcc-default64;
sed -e "s/@DATE@/`date`/g" src/Makefile.in >src/Makefile
(cd src; make)
make[1]: Entering directory '/mnt/work/ghorvath/gap4r8/pkg/ace-5.2/src'
gcc -Wall -pedantic -ansi -O2 -DAL2_HINFO -DAL0_STAT -DAL0_CC -DAL0_DD -DDATE="\""'2016. dec. 13., kedd, 10.39.15 CET'"\"" -o ace \
        ace.c parser.c postproc.c util2.c control.c util1.c coinc.c enum.c util0.c
parser.c: In function ‘al2_cmdloop’:
parser.c:1806:7: warning: ‘f’ may be used uninitialized in this function [-Wmaybe-uninitialized]
       al1_prtct(intarr[0], intarr[1], intarr[2], FALSE, f);
       ^
coinc.c: In function ‘al0_cols12’:
coinc.c:79:37: warning: ‘high2s’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     else if (low2s == clo && high2s == chi) \
                                     ^
coinc.c:240:29: note: ‘high2s’ was declared here
   int low1s, low2s, high1s, high2s; /* consequent coincidences */
                             ^
coinc.c:77:32: warning: ‘high1s’ may be used uninitialized in this function [-Wmaybe-uninitialized]
     if (low1s == clo && high1s == chi)      \
                                ^
coinc.c:240:21: note: ‘high1s’ was declared here
   int low1s, low2s, high1s, high2s; /* consequent coincidences */
                     ^
enum.c: In function ‘al0_cl’:
enum.c:394:6: warning: ‘k’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   if (k == 0)
      ^
enum.c: In function ‘al0_rdefn’:
enum.c:524:6: warning: ‘l’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   if (l == 0)
      ^
enum.c: In function ‘al0_rl.constprop’:
enum.c:295:6: warning: ‘l’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   if (l == 0)
      ^
In file included from enum.c:1015:0:
enum.c: In function ‘al0_enum’:
enum01.c:118:6: warning: ‘l’ may be used uninitialized in this function [-Wmaybe-uninitialized]
   if (l == 0)
      ^
In file included from enum.c:1015:0:
enum01.c:47:48: note: ‘l’ was declared here
   int first, last, i, ii, j, ifront, iback, k, l, m, mi, n;
                                                ^
enum.c:1619:12: warning: ‘isave’ may be used uninitialized in this function [-Wmaybe-uninitialized]
         if ((result = al0_apply(1,pj,pk,TRUE,isave)) >= 0)
            ^
make[1]: Leaving directory '/mnt/work/ghorvath/gap4r8/pkg/ace-5.2/src'
mv src/ace bin/x86_64-pc-linux-gnu-gcc-default64/ace
````
Out of these, only the following go to ````stdout```` (all the rest go to ````stderr````):
````
==== Checking ace-5.2
../../sysinfo.gap found
Makefile successfully created.
Done. Now call 'make'
mkdir -p bin/x86_64-pc-linux-gnu-gcc-default64;
sed -e "s/@DATE@/`date`/g" src/Makefile.in >src/Makefile
(cd src; make)
make[1]: Entering directory '/mnt/work/ghorvath/gap4r8/pkg/ace-5.2/src'
gcc -Wall -pedantic -ansi -O2 -DAL2_HINFO -DAL0_STAT -DAL0_CC -DAL0_DD -DDATE="\""'2016. dec. 13., kedd, 10.39.15 CET'"\"" -o ace \
        ace.c parser.c postproc.c util2.c control.c util1.c coinc.c enum.c util0.c
make[1]: Leaving directory '/mnt/work/ghorvath/gap4r8/pkg/ace-5.2/src'
mv src/ace bin/x86_64-pc-linux-gnu-gcc-default64/ace
````

**Comment by fingolfin** [comment]
I actually got none of these warnings with my default compiler (clang 3.9). But I do get warnings with GCC 5.4, and will now also install GCC 6.2.

In the meantime, most of these warnings look harmless.

**Comment by hungaborhorvath** [comment]
If it matters, I am on debian stable (currently jessie) which shipped with gcc 4.9.
